packages/expo-font/ios/FontLoaderModule.swift

```diff
@@ -1,13 +1,34 @@
 import ExpoModulesCore
 
 public final class FontLoaderModule: Module {
+  private let customNativeFonts: [String]
+  // could be a Set, but to be able to pass to JS we keep it as an array
+  private var registeredFonts: [String]
+
+  public required init(appContext: AppContext) {
+    let nativeFonts = queryCustomNativeFonts()
+    self.customNativeFonts = nativeFonts
+    self.registeredFonts = nativeFonts
+    super.init(appContext: appContext)
+  }
+
   public func definition() -> ModuleDefinition {
     Name("ExpoFontLoader")
 
-    Property("customNativeFonts") {
-      return queryCustomNativeFonts()
+    Constants([
+      "customNativeFonts": self.customNativeFonts
+    ])
+
+    // NOTE: this is exposed in JS as globalThis.expo.modules.ExpoFontLoader.loadedFonts
+    // and potentially consumed outside of Expo (e.g. RN vector icons)
+    // do NOT change the property as it'll break consumers!
```

Comment:
ditto about potentially not needing this comment or the one below

Reply:
let's discuss in https://github.com/expo/expo/pull/30431/files#r1684587690